pkg/ddl/notifier/subscribe.go

```diff
@@ -268,15 +268,20 @@ func (n *DDLNotifier) processEventForHandler(
 	if (change.processedByFlag & (1 << handlerID)) != 0 {
 		return nil
 	}
+	newFlag := change.processedByFlag | (1 << handlerID)
 
-	if err = session.Begin(ctx); err != nil {
+	if err = session.BeginPessimistic(ctx); err != nil {
```

Comment:
what's the target issue of using pessimistic txn ?

Reply:
in pessimistic txn, errors can be found earlier, such as when execute the SQL. So handler can see the error and process it. Otherwise the error is reported when COMMIT, and in current handler interface it can't be proccessed

Reply:
> in pessimistic txn, errors can be found earlier

if there is no check about `affected rows`, there is no error actually in pessimistic txn

Reply:
The original problem is handler uses the txn to insert some duplicate rows. In optimistic txn, there's no error after execute the INSERT. Using pessimistic txn handler can see "duplicate entry" error. https://github.com/pingcap/tidb/pull/58980 is related

Reply:
I also checked the affected rows in https://github.com/pingcap/tidb/pull/59157/commits/bef13bf134d125a8db59f82a0f91e629052d7eda . It's revealed by old UT after switching to pessimistic txn.